modules/tinymce/src/themes/silver/test/ts/browser/editor/ToolbarPersistTest.ts

```diff
@@ -0,0 +1,83 @@
+import { Chain, Log, NamedChain, Pipeline, Assertions, ApproxStructure, Waiter } from '@ephox/agar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { Editor as McEditor } from '@ephox/mcagar';
+import { SugarElement, SugarBody, Insert, Focus, Remove } from '@ephox/sugar';
+import Theme from 'tinymce/themes/silver/Theme';
+import Editor from 'tinymce/core/api/Editor';
+
+UnitTest.asynctest('Toolbar persist test', (success, failure) => {
+  Theme();
+
+  const cWaitForEditorVisibility = (message: string, visible: boolean) => Waiter.cTryUntil(message, Chain.op((editor: Editor) => {
+    Assertions.assertStructure(
+      message,
+      ApproxStructure.build((s, str, arr) => s.element('div', {
+        classes: [ arr.has('tox-tinymce-inline') ],
+        attrs: {
+          style: str.contains('display: ' + (visible ? 'flex' : 'none'))
+        }
+      })),
+      SugarElement.fromDom(editor.getContainer())
+    );
+  }));
+
+  const cShowEditor = Chain.fromChains([
+    Chain.op((editor: Editor) => editor.ui.show()),
+    cWaitForEditorVisibility('Wait for editor to be shown', true)
```

Comment:
Every time you call cShowEditor or cHideEditor you follow it up by asserting visibility anyway. Do you need to check in here as well?

Reply:
Yeah that shouldn't be duplicated